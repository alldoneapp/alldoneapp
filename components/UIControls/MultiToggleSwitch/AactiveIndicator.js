import React, { useEffect, useRef, useState } from 'react'
import { Animated, StyleSheet } from 'react-native'
import { useSelector } from 'react-redux'

import MyPlatform from '../../MyPlatform'
import { colors } from '../../styles/global'

export default function AactiveIndicator({ options, optionsRefs, currentIndex }) {
    const smallScreenNavigation = useSelector(state => state.smallScreenNavigation)
    const [optionsWidths, setOptionsWidths] = useState(new Array(options.length).fill(0))
    const width = useRef(new Animated.Value(0)).current
    const position = useRef(new Animated.Value(0)).current

    const getOffset = () => {
        let width = 0
        for (let i = 0; i < currentIndex; i++) {
            width += optionsWidths[i]
        }

        return width
    }

    const animate = widths => {
        let posValue = getOffset()

        Animated.parallel(
            [
                Animated.timing(position, {
                    toValue: posValue,
                    duration: 300,
                }),
                Animated.timing(width, {
                    toValue: widths[currentIndex],
                    duration: 200,
                }),
            ],
            { stopTogether: false }
        ).start()
    }

    const updateWidths = async () => {
        let functors = []

        for (let i = 0; i < optionsRefs.length; i++) {
            functors.push(MyPlatform.getElementWidth(optionsRefs[i]))
        }

        const widths = await Promise.all(functors)

        animate(widths)
        setOptionsWidths(widths)
    }

    useEffect(() => {
        updateWidths()
    }, [currentIndex])

    useEffect(() => {
        updateWidths()
    }, [JSON.stringify(optionsWidths)])

    useEffect(() => {
        updateWidths()
    }, [smallScreenNavigation])

    useEffect(() => {
        setOptionsWidths(new Array(options.length).fill(0))
    }, [JSON.stringify(options)])

    return (
        <Animated.View style={[localStyles.activeIndicator, { width: width, transform: [{ translateX: position }] }]} />
    )
}

const localStyles = StyleSheet.create({
    activeIndicator: {
        position: 'absolute',
        height: 22,
        width: 50,
        backgroundColor: colors.Grey100,
        borderRadius: 12,
        shadowColor: colors.Text03,
        shadowOffset: { width: 0, height: 0 },
        shadowOpacity: 1,
        shadowRadius: 8,
        elevation: 3,
    },
})
