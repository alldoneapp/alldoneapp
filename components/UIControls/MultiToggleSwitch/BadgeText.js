import React from 'react'
import { Text, StyleSheet } from 'react-native'
import { useSelector } from 'react-redux'

import styles, { colors } from '../../styles/global'

export default function BadgeText({ active, badge }) {
    const smallScreenNavigation = useSelector(state => state.smallScreenNavigation)
    return (
        <Text
            style={[
                localStyles.text,
                { color: active ? colors.Primary100 : colors.Text03 },
                smallScreenNavigation && !active ? { marginLeft: 4 } : { paddingRight: 12 },
            ]}
        >
            {badge}
        </Text>
    )
}

const localStyles = StyleSheet.create({
    text: {
        ...styles.caption2,
        fontSize: 14,
        lineHeight: 22,
        height: 22,
        minHeight: 22,
        maxHeight: 22,
    },
})
