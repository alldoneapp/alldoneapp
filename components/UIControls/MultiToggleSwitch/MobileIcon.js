import React from 'react'
import { useSelector } from 'react-redux'

import Icon from '../../Icon'
import { colors } from '../../styles/global'

export default function MobileIcon({ icon, active }) {
    const smallScreenNavigation = useSelector(state => state.smallScreenNavigation)

    return (
        <Icon
            name={icon}
            size={20}
            color={active ? colors.Primary100 : colors.Text03}
            style={(!smallScreenNavigation || active) && { marginLeft: 8 }}
        />
    )
}
