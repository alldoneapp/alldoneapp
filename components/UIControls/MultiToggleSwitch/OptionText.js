import React from 'react'
import { Text, StyleSheet } from 'react-native'
import { useSelector } from 'react-redux'

import { translate } from '../../../i18n/TranslationService'
import styles, { colors } from '../../styles/global'

export default function OptionText({ active, badge, text }) {
    const smallScreenNavigation = useSelector(state => state.smallScreenNavigation)

    return (
        <Text
            style={[
                localStyles.optionText,
                badge == null && localStyles.optionTextWithoutBadge,
                active && localStyles.activeIndex,
                smallScreenNavigation && { paddingLeft: 4 },
            ]}
        >
            {translate(text)}
        </Text>
    )
}

const localStyles = StyleSheet.create({
    optionText: {
        ...styles.body2,
        color: colors.Text03,
        justifyContent: 'center',
        alignItems: 'center',
        height: 22,
        paddingLeft: 12,
        paddingRight: 4,
        fontWeight: '400',
    },
    optionTextWithoutBadge: {
        paddingRight: 12,
    },
    activeIndex: {
        ...styles.subtitle2,
        color: colors.Primary100,
    },
})
