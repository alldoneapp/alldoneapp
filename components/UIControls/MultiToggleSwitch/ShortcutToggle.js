import React from 'react'
import Hotkeys from 'react-hot-keys'
import { useSelector } from 'react-redux'

export default function ShortcutToggle({ currentIndex, options, onSelectOption }) {
    const blockShortcuts = useSelector(state => state.blockShortcuts)

    const onShortcutToggle = () => {
        const indexes = []
        for (let i = 0; i < options.length; i++) {
            if (options[i]) indexes.push({ index: i, option: options[i] })
        }

        const index = currentIndex === options.length - 1 ? indexes[0] : indexes[currentIndex + 1]
        onSelectOption(index.index, index.option.text)
    }
    return <Hotkeys disabled={blockShortcuts} keyName={'alt+G'} onKeyDown={onShortcutToggle} filter={e => true} />
}
