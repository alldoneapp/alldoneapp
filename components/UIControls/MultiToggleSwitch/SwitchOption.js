import React from 'react'
import { StyleSheet, TouchableOpacity, View } from 'react-native'
import { useSelector } from 'react-redux'

import BadgeText from './BadgeText'
import MobileIcon from './MobileIcon'
import OptionText from './OptionText'

export default function SwitchOption({ i, optionsRefs, onSelectOption, active, option }) {
    const smallScreenNavigation = useSelector(state => state.smallScreenNavigation)
    const { text, icon, badge } = option
    return (
        <TouchableOpacity
            ref={ref => {
                optionsRefs[i] = ref
            }}
            onPress={() => {
                onSelectOption(i, text)
            }}
            style={localStyles.optionIconTouch}
        >
            <View style={[localStyles.optionTextContainer, smallScreenNavigation && !active && localStyles.optionIcon]}>
                {smallScreenNavigation && <MobileIcon icon={icon} active={active} />}
                {(!smallScreenNavigation || active) && <OptionText active={active} badge={badge} text={text} />}
                {badge != null && <BadgeText active={active} badge={badge} />}
            </View>
        </TouchableOpacity>
    )
}

const localStyles = StyleSheet.create({
    optionIconTouch: {
        justifyContent: 'center',
        alignItems: 'center',
    },
    optionIcon: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        paddingLeft: 8,
        paddingRight: 8,
    },
    optionTextContainer: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    },
})
