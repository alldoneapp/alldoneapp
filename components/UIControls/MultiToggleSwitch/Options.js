import React from 'react'
import { StyleSheet, View } from 'react-native'

import SwitchOption from './SwitchOption'

export default function Options({ optionsRefs, onSelectOption, currentIndex, options }) {
    return (
        <View style={localStyles.optionsContainer}>
            {options.map((option, i) => {
                return (
                    option && (
                        <SwitchOption
                            key={i}
                            i={i}
                            optionsRefs={optionsRefs}
                            onSelectOption={onSelectOption}
                            active={currentIndex === i}
                            option={option}
                        />
                    )
                )
            })}
        </View>
    )
}

const localStyles = StyleSheet.create({
    optionsContainer: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    },
})
