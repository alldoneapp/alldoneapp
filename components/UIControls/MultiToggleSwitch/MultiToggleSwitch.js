import React, { useRef, useEffect } from 'react'
import { StyleSheet, View } from 'react-native'
import { useSelector } from 'react-redux'

import { colors } from '../../styles/global'
import Shortcut from '../Shortcut'
import Options from './Options'
import ShortcutToggle from './ShortcutToggle'
import AactiveIndicator from './AactiveIndicator'

export default function MultiToggleSwitch({ currentIndex, options, onChangeOption, containerStyle }) {
    const showShortcuts = useSelector(state => state.showShortcuts)
    const showFloatPopup = useSelector(state => state.showFloatPopup)

    const optionsRefs = useRef(options.map(_ => null))

    const onSelectOption = (index, optionText) => {
        if (onChangeOption) onChangeOption(index, optionText)
    }

    useEffect(() => {
        optionsRefs.current = options.map(_ => null)
    }, [JSON.stringify(options)])

    return (
        <View style={containerStyle}>
            <View style={localStyles.backContainer}>
                <ShortcutToggle currentIndex={currentIndex} options={options} onSelectOption={onSelectOption} />
                <AactiveIndicator optionsRefs={optionsRefs.current} currentIndex={currentIndex} options={options} />
                <Options
                    optionsRefs={optionsRefs.current}
                    onSelectOption={onSelectOption}
                    currentIndex={currentIndex}
                    options={options}
                />
            </View>
            {showShortcuts && showFloatPopup === 0 && (
                <Shortcut text={'G'} containerStyle={[{ position: 'absolute', top: -4, right: 0 }]} />
            )}
        </View>
    )
}

MultiToggleSwitch.defaultProps = {
    currentIndex: 0,
}

const localStyles = StyleSheet.create({
    backContainer: {
        height: 26,
        position: 'relative',
        alignSelf: 'baseline',
        paddingTop: 2,
        paddingBottom: 2,
        paddingLeft: 2,
        paddingRight: 2,
        flexDirection: 'row',
        backgroundColor: colors.Grey300,
        borderRadius: 14,
        overflow: 'hidden',
    },
})
