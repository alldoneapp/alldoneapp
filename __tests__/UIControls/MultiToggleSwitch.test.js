import React from 'react'
import MultiToggleSwitch from '../../components/UIControls/MultiToggleSwitch'

import renderer from 'react-test-renderer'

jest.mock('firebase', () => ({ firestore: {} }));

let options = [
    { icon: 'square', text: 'Open' },
    { icon: 'clock', text: 'Pending' },
    { icon: 'square-checked-gray', text: 'Done' },
]

describe('MultiToggleSwitch component', () => {
    describe('MultiToggleSwitch empty snapshot test', () => {
        it('Should render correctly', () => {
            const tree = renderer.create(<MultiToggleSwitch options={options} />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })

    describe('Test component functions', () => {
        it('after call componentDidMount should render correctly', () => {
            const tree = renderer.create(<MultiToggleSwitch options={options} />)
            const instance = tree.getInstance()
            instance.componentDidMount()
        })
        it('after call componentDidUpdate should render correctly', () => {
            const tree = renderer.create(<MultiToggleSwitch options={options} />)
            const instance = tree.getInstance()
            instance.componentDidUpdate()
        })
        it('after call updateWidths should render correctly', () => {
            const tree = renderer.create(<MultiToggleSwitch options={options} />)
            const instance = tree.getInstance()
            instance.updateWidths()

            // Waiting to resolve the promise
            setTimeout(() => {
                let state = instance.state
                expect(state.optionsWidths.length).toEqual(3)
            }, 50)
        })
        xit('after call onSelectOption should render correctly', () => {
            const tree = renderer.create(<MultiToggleSwitch options={options} onChangeOption={() => {}} />)
            const instance = tree.getInstance()
            instance.onSelectOption(1, 'text')

            // Waiting to resolve the promise
            // Even with a timeout of 1s it is not taking the value for activeIndex.
            setTimeout(() => {
                let state = instance.state
                expect(state.activeIndex).toEqual(1)
            }, 1000)
        })
        it('after call animate should render correctly', () => {
            const tree = renderer.create(<MultiToggleSwitch options={options} />)
            const instance = tree.getInstance()
            instance.animate()
        })
        it('after call getOffset should render correctly', () => {
            const tree = renderer.create(<MultiToggleSwitch options={options} />)
            const instance = tree.getInstance()
            instance.state.optionsWidths = [10, 10, 10]
            let offset = instance.getOffset(1)
            expect(offset).toEqual(10)
        })
        it('after call updateState should render correctly', () => {
            const tree = renderer.create(<MultiToggleSwitch options={options} />)
            expect(tree.toJSON()).toMatchSnapshot()

            tree.getInstance().updateState()
            expect(tree.toJSON()).toMatchSnapshot()
        })
        it('after call componentWillUnmount should render correctly', () => {
            const tree = renderer.create(<MultiToggleSwitch options={options} />)
            tree.getInstance().componentWillUnmount()
        })
    })
})
