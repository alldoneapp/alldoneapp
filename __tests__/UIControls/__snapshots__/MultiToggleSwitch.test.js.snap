// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiToggleSwitch component MultiToggleSwitch empty snapshot test Should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignSelf": "baseline",
      "backgroundColor": "#E7ECEF",
      "borderRadius": 14,
      "flexDirection": "row",
      "height": 26,
      "overflow": "hidden",
      "paddingBottom": 2,
      "paddingLeft": 2,
      "paddingRight": 2,
      "paddingTop": 2,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FAFBFB",
        "borderRadius": 12,
        "elevation": 3,
        "height": 22,
        "position": "absolute",
        "shadowColor": "#8A94A6",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 8,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#007FFF",
              "fontFamily": "Roboto-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "height": 22,
              "justifyContent": "center",
              "letterSpacing": 0.16,
              "lineHeight": 22,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          Open
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#8A94A6",
              "fontFamily": "Roboto-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "height": 22,
              "justifyContent": "center",
              "letterSpacing": 0.32,
              "lineHeight": 22,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          Pending
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#8A94A6",
              "fontFamily": "Roboto-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "height": 22,
              "justifyContent": "center",
              "letterSpacing": 0.32,
              "lineHeight": 22,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          Done
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`MultiToggleSwitch component Test component functions after call updateState should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignSelf": "baseline",
      "backgroundColor": "#E7ECEF",
      "borderRadius": 14,
      "flexDirection": "row",
      "height": 26,
      "overflow": "hidden",
      "paddingBottom": 2,
      "paddingLeft": 2,
      "paddingRight": 2,
      "paddingTop": 2,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FAFBFB",
        "borderRadius": 12,
        "elevation": 3,
        "height": 22,
        "position": "absolute",
        "shadowColor": "#8A94A6",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 8,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#007FFF",
              "fontFamily": "Roboto-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "height": 22,
              "justifyContent": "center",
              "letterSpacing": 0.16,
              "lineHeight": 22,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          Open
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#8A94A6",
              "fontFamily": "Roboto-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "height": 22,
              "justifyContent": "center",
              "letterSpacing": 0.32,
              "lineHeight": 22,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          Pending
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#8A94A6",
              "fontFamily": "Roboto-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "height": 22,
              "justifyContent": "center",
              "letterSpacing": 0.32,
              "lineHeight": 22,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          Done
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`MultiToggleSwitch component Test component functions after call updateState should render correctly 2`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignSelf": "baseline",
      "backgroundColor": "#E7ECEF",
      "borderRadius": 14,
      "flexDirection": "row",
      "height": 26,
      "overflow": "hidden",
      "paddingBottom": 2,
      "paddingLeft": 2,
      "paddingRight": 2,
      "paddingTop": 2,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FAFBFB",
        "borderRadius": 12,
        "elevation": 3,
        "height": 22,
        "position": "absolute",
        "shadowColor": "#8A94A6",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 8,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#007FFF",
              "fontFamily": "Roboto-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "height": 22,
              "justifyContent": "center",
              "letterSpacing": 0.16,
              "lineHeight": 22,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          Open
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#8A94A6",
              "fontFamily": "Roboto-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "height": 22,
              "justifyContent": "center",
              "letterSpacing": 0.32,
              "lineHeight": 22,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          Pending
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "center",
              "color": "#8A94A6",
              "fontFamily": "Roboto-Regular",
              "fontSize": 14,
              "fontWeight": "400",
              "height": 22,
              "justifyContent": "center",
              "letterSpacing": 0.32,
              "lineHeight": 22,
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          Done
        </Text>
      </View>
    </View>
  </View>
</View>
`;
